test/topology/test_mutation_schema_change.py

```diff
@@ -0,0 +1,158 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""
+Reproducer for a failure during lwt operation due to missing of a column mapping in schema history table.
+"""
+import asyncio
+import logging
+import time
+from functools import partial
+from test.pylib.rest_client import inject_error_one_shot, inject_error
+from test.pylib.util import wait_for
+from test.pylib.manager_client import ManagerClient
+from test.pylib.internal_types import IPAddress, ServerInfo
+import pytest
+from cassandra.cluster import Cluster, ConsistencyLevel  # type: ignore # pylint: disable=no-name-in-module
+from cassandra.query import SimpleStatement              # type: ignore # pylint: disable=no-name-in-module
+
+
+logger = logging.getLogger(__name__)
+
+
+async def server_sees_another_server(server: ServerInfo, manager: ManagerClient):
+    alive_nodes = await manager.api.get_alive_endpoints(server.ip_addr)
+    if len(alive_nodes) > 1:
+        return True
+
+@pytest.mark.asyncio
+async def test_mutation_schema_change(manager, random_tables):
+    """
+        Cluster A, B, C
+        create table
+        stop C
+        change schema + do lwt write + change schema
+        stop B
+        start C
+        do lwt write to the same key through C
+    """
+    server_a, server_b, server_c = await manager.running_servers()
+    t = await random_tables.add_table(ncolumns=5, pks=1)
+    manager.driver_close()
+    # Reduce the snapshot thresholds
+    await manager.mark_dirty()
+    errs = [inject_error_one_shot(manager.api, s.ip_addr, 'raft_server_snapshot_reduce_threshold')
+            for s in [server_a, server_b, server_c]]
+    await asyncio.gather(*errs)
+
+
+    logger.info("Stopping C %s", server_c)
+    await manager.server_stop_gracefully(server_c.server_id)
+    await manager.driver_connect()
+
+    async with inject_error(manager.api, server_b.ip_addr, 'paxos_error_before_learn'):
+        await t.add_column()
+        ROWS = 1
+        seeds = [t.next_seq() for _ in range(ROWS)]
+        stmt = f"INSERT INTO {t} ({','.join(c.name for c in t.columns)}) " \
+               f"VALUES ({', '.join(['%s'] * len(t.columns))}) "           \
+               f"IF NOT EXISTS"
+        query = SimpleStatement(stmt, consistency_level=ConsistencyLevel.ONE)
+        for seed in seeds:
+            logger.info("INSERT row seed %s", seed)
+            await manager.cql.run_async(query, parameters=[c.val(seed) for c in t.columns])
+        await t.add_column()
+
+    manager.driver_close()
+
+    logger.info("Stopping B %s", server_b)
+    await manager.server_stop_gracefully(server_b.server_id)
+    logger.info("Starting C %s", server_c)
+    await manager.server_start(server_c.server_id)
+
+    await wait_for(partial(server_sees_another_server, server_c, manager), time.time() + 45, period=.1)
```

Comment:
I don't understand why here you wait only for C to see another server, but in the other test you wait for both servers to see each other.

Reply:
The query here goes to C so it needs to know at least of 1 more to have QUORUM.

In the other location it goes through A so it needs to wait for it to know of 1 more for QUORUM. Perhaps the wait for C in the second case is not needed.